### ISSUE TITLE
[5.1] Support a "legacy" options UI with new modifier framework

### ISSUE THREAD

**Issue by ListenerJubatus** [stepmania/stepmania#1331]
Currently, the NestyMenu is the only UI capable of utilizing the new modifier system. Its design, however, has been polarizing and controversal.

An alternative options UI should be coded that is more in line with the "classic", row-based layout, as an option for use in themes.

**Comment by SpoOkyMagician** [comment]
Eww... No way. I like the nesty version. It's simple, to the point, and it works just fine. However, I am okay with adding a option for it if Kyzentun is...

**Comment by dguzek** [comment]
In my opinion, the NestyMenu is good.  It's (as far as I can tell) flexible and allows themers to compartmentalize their PlayerOptions, should they want to.  I don't think the NestyMenu, as a framework, has been polarizing and controversial, and I think that it should probably stay, albeit maybe with more documentation.

The new system for setting numeric values is where I "take issue," because it requires end-users to load and manage a more complex algorithm into their cognitive space, so to speak.  

I'll use speed modifier as an example, but the following pseudo-code should be generally applicable to any numeric option via a Nesty sub-menu.

The mental algorithm for setting a speed mod using an old-school OptionRow might look something like this:

```
function SetSpeedMod(){
	if (I want to increment my SpeedMod){
		Press MenuRight
	
		if (SpeedMod is where I want it){
			return
		} else {
			IncrementSpeedMod()
		}
	} else if (I want to decrement my SpeedMod){
		// ...
	}
}
```

The mental algorithm doing the same using NestyMenu's sub-menu for setting numeric values might look like this:

```
function SetSpeedMod(){
	if (I want to increment my SpeedMod){
		if (my desired SpeedMod > 1000){
			while (selection != "+1000")
				Press MenuRight
		} else if (my desired SpeedMod > 100 and my desired SpeedMod < 1000){
			while (selection != "+100")
				Press MenuRight
		} else if (my desired SpeedMod > 10 and my desired SpeedMod < 100){
			while (selection != "+10")
				Press MenuRight
		} else if (my desired SpeedMod > 1 and my desired SpeedMod < 10){
			while (selection != "+1")
				Press MenuRight		
		}

		Press Enter
	
		if (SpeedMod is where I want it){
			return
		} else {
			SetSpeedMod()
		}
	} else if ( I want to decrement my SpeedMod){
		//...
	}
}
```

Basically, I don't have the mental RAM to easily operate that particular sub-menu, and my informal testing (using Consensual periodically on the local SM5 arcade machine) with novice and expert users suggests I'm not alone in this.

**Comment by dguzek** [comment]
With all that said, I think more options are better than fewer.  

It is very plausible that people more intelligent than I have no difficulty managing the `+1000/-1000/+100/-100/+10/-10/+1/-1` system, and I think that there's no reason to discard it outright, which I think @SpoOkyMagician seems to fear might be the case.

**Comment by ListenerJubatus** [comment]
Note that I am not intending to outright replace NestyMenu, but provide an option for those wanting to simulate/create a more "traditional" UI (or re-create one which does not use it) in their themes. DDR A has a rather crazy options menu design that otherwise functions like the "classic" menu but _almost_ looks like the current nestymenu. It might be possible to get close to something like this with such a system.

![](http://i.imgur.com/wjzpxy5.png)

But on that second note, I think re-arranging speed mods to have positive units at the top, and negative units at the bottom, and maybe changing the Xmod increments to 1.0x 0.5x and 0.1x (a bit more likely than 10, 1, 0.1) would make it easier to get to common speeds such as 1.5x

**Comment by kyzentun** [comment]
[![](http://i.imgur.com/GPeojxl.jpg)](http://kyzentun.tejat.net/rant)

**Comment by dguzek** [comment]
Since Simply Love is the only theme I know of with a "PlayerOptions2" screen, I'll assume @kyzentun is indirectly addressing me.  If this is not the case, I apologize.

I disagree with the original issue as phrased by @ListenerJubatus.  I don't have reason to believe that the NestyMenu was controversial or polarizing (or a bad thing).  

As I stated above:
> In my opinion, the NestyMenu is good. It's (as far as I can tell) flexible and allows themers to compartmentalize their PlayerOptions, should they want to. I don't think the NestyMenu, as a framework, has been polarizing and controversial, and I think that it should probably stay, albeit maybe with more documentation.
>
> The new system for setting numeric values is where I "take issue," because it requires end-users to load and manage a more complex algorithm into their working memory, so to speak.

**Comment by kyzentun** [comment]
Version short enough for you to read:  I'm not writing lua option rows for it because Option Row is garbage (and passing multiple args to a lua option row function doesn't work anyway).  Nobody else is going to write lua option rows because they can't figure out the prefs system.

So this issue is pointless.

http://kyzentun.tejat.net/rant
